### PR TITLE
New tests

### PR DESCRIPTION
**Pull request by craigmeyer**
- GPU-MPI tests
- GPU compilation (performance-modelling tools with HIP, OMP, ACC, HIP+OMP, HIP+ACC) and associated performance tests
- SLURM CPU tests (including memory assignment, billing, affinity)
- SLURM GPU tests (including affinity, GPU/CPU assignment

### CODE REVIEW

File: common/performance-modelling-tools/tests/README.md
Comment (pelahi) [inline common/performance-modelling-tools/tests/README.md:1]: Can you flesh out the read me a bit more.

**Review by pelahi — COMMENTED** [review]
I think this looks like a great addition to the reframe tests. I think we'll need to change the repo name and description but let's merge all of this soon. My only request is if you can also update the main README as well.

**Comment by craigmeyer** [comment]
Yes, addressing the README situation was next on my list. I will make sure to update the main README of the repo and that of performance_modelling_tools.

I was actually thinking of doing a README for each subset of tests as well (MPI, SLURM, compilation), to briefly explain them and what users might need to change or consider for use on their own system. What do you think?

**Comment by pelahi** [comment]
Yes, I think it's a great idea for each subset of tests to have a README. Will make this repo more widely useful.

**Comment by craigmeyer** [comment]
I've updated the PR to now include READMEs and SLURM sacct accounting tests.

**Review by pelahi — APPROVED** [review]

**Pull request merged**